### ISSUE TITLE
RDATA_SIZE's size

### ISSUE THREAD

**Issue by Licenser** [dalmatinerdb/mmath#5]
@szarsti I'm trying to figure out why ?RDATA_SIZE is defined as 16 byte, from the c code it looks like the plain data structure should be 16*6 9 byte (8 for the coefficient + 1 for the exponent).

**Comment by Licenser** [comment]
answered my own question, it has to d with how c packs structs, it seems to do so in 64 byte chunks at least on a 64 byte system

**Comment by szarsti** [comment]
Yes, you find right answer.

I want to replace it with a function, so it can be safe to use on different architectures. Current solution works only on 64bit machines. This is last remaining thing I want to do before preparing pull request with floating point decimals support.

**Comment by Licenser** [comment]
Something like ` __attribute__((__packed__))` might be an option that should ™️ reduce it to the actual size of the content, or we fill up the content with other data to align with 64 byte, at the certenty branch there are 32 more bit used in there leaving 'only' 24 bit free

**Comment by szarsti** [comment]
Good idea, especially if we add another field for confidence. We could use 32bit for it, so all of them fits into 16 bits (using 32bit exponent).

**Comment by Licenser** [comment]
sorry I was meaning confidence not certenty, I keep mixing the two up
